### PR TITLE
587 asset group deletion corrections

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Asset deletion from asset group done as cascade to prevent duplicate sighting deletion
- EDM propogation of delete of sighting

---

**Review Notes**
- Specify a list of any items, notes, or qualifiers that need to be known in order to efficiently review this PR

**Model Updates [OPTIONAL]**

`ExampleModel`
- `guid`
- `example_attribute`

**Database Updates [OPTIONAL]**

Specify any new/modified/removed database tables

**REST API Updates [OPTIONAL]**

A new example POST API is defined with its expected parameters, if needed to be specified for a thorough review

```
[POST] /api/v1/module/endpoint/
{
    'example_attribute': 'example',
}
```

**Invoke CLI Updates [OPTIONAL]**

Here is an example invoke that shows the new functionality

```
$ invoke app.module.task
<copy relevant terminal output>
```

### CODE REVIEW

File: app/modules/assets/models.py [hunk 1]
```diff
@@ -285,14 +285,19 @@ def get_or_make_master_format_path(self):
             rgb.save(target_path)
         return target_path
 
-    def delete(self):
-        asset_group = self.asset_group
+    # Delete of an asset as part of deletion of asset_group
```
Comment (hwindsor) [inline app/modules/assets/models.py:288]: This was required to avoid asset group double deletion

File: app/modules/sightings/resources.py [hunk 2]
```diff
@@ -614,48 +613,14 @@ def delete(self, sighting):
                 code=400,
             )
 
-        response_data = None
-        if response.ok:
-            response_data = response.json()
-
-        if not response.ok or not response_data.get('success', False):
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:621]: These would never have been hit as this scenario results in a houston exception which is handled above

File: app/modules/assets/models.py [hunk 3]
```diff
@@ -285,14 +285,19 @@ def get_or_make_master_format_path(self):
...
             for sighting in self.asset_sightings:
                 db.session.delete(sighting)
             db.session.delete(self)
+
+    # delete not part of asset group deletion so must inform asset group that we're gone
+    def delete(self):
+        asset_group = self.asset_group
+        self.delete_cascade()
         db.session.refresh(asset_group)
         asset_group.justify_existence()
```
Comment (colinwkingen) [inline app/modules/assets/models.py:302]: Still one of my favorite method names.

**Review by colinwkingen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**